File: 10 - 11. Sorting and Searching/code/linear_search.py
def linear_search(arr, x):
    for i in range(len(arr)):
        if arr[i] == x:
            return i
    return -1

if __name__ == "__main__":
    arr = [10, 50, 30, 70, 80, 60, 20, 90, 40]
    key = 30
    
    result = linear_search(arr, key)
    
    if result != -1:
        print("Element found at index:", result)
    else:
        print("Element not found in the array")

File: 10 - 11. Sorting and Searching/code/linear_search.cpp
#include <iostream>
#include <vector>
using namespace std;

int linearSearch(vector<int> &arr, int key) {
    int n = arr.size();
    for (int i = 0; i < n; i++) {
        if (arr[i] == key) {
            return i;
        }
    }
    return -1;
}

int main() {
    vector<int> arr = {10, 50, 30, 70, 80, 60, 20, 90, 40};
    int key = 30;

    int index = linearSearch(arr, key);

    if (index != -1) {
        cout << "Element found at index: " << index << endl;
    } else {
        cout << "Element not found" << endl;
    }

    return 0;
}